subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/dependencies/ModuleDependenciesGraphPlugin.kt

```diff
@@ -0,0 +1,24 @@
+package com.avito.module.dependencies
+
+import com.avito.module.internal.dependencies.TestPlugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+@Suppress("UnstableApiUsage")
+public class ModuleDependenciesGraphPlugin : Plugin<Project> {
+
+    override fun apply(target: Project) {
+        require(target.rootProject == target) {
+            "must be applied to the root project"
+        }
+        val enableTestPlugin = target.providers
+            .gradleProperty("injected.from.gradle_testkit")
+            .forUseAtConfigurationTime()
+            .map { it.toBoolean() }
+            .getOrElse(false)
+        if (enableTestPlugin) {
```

Comment:
I decide to make a separate TestPlugin which I apply if `injected.from.gradle_testkit` is present
`